auth/tron-signer.go

```diff
@@ -5,16 +5,34 @@ package auth
 import (
 	"crypto/ecdsa"
 
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/loomnetwork/go-loom/common/evmcompat"
 	sha3 "github.com/miguelmota/go-solidity-sha3"
 )
 
 type TronSigner struct {
 	PrivateKey *ecdsa.PrivateKey
 }
 
+func NewTronSigner(privateKey []byte) *TronSigner {
+	if privateKey == nil {
+		privKey, err := btcec.NewPrivateKey(btcec.S256())
+		if err != nil {
+			panic(err)
+		}
+		return &TronSigner{PrivateKey: privKey.ToECDSA()}
+	}
+
+	privKey, _ := btcec.PrivKeyFromBytes(btcec.S256(), common.FromHex(string(privateKey)))
+	return &TronSigner{PrivateKey: privKey.ToECDSA()}
+}
+
 func (k *TronSigner) Sign(txBytes []byte) []byte {
-	signature, err := crypto.Sign(sha3.SoliditySHA3(txBytes), k.PrivateKey)
+	signature, err := evmcompat.GenerateTypedSig(
+		sha3.SoliditySHA3(txBytes), k.PrivateKey, evmcompat.SignatureType_EIP712,
```

Comment:
So in loom-js @tstxni97 used `3` not `0` for the sig type https://github.com/loomnetwork/loom-js/pull/268/files#diff-d53e0b623e5d95fd76ae5f1121c4baabR31 aka `SignatureType_TRON`

Reply:
fixed